packages/amplify_datastore/ios/Classes/types/model/FlutterSerializedModel.swift

```diff
@@ -235,44 +210,40 @@ struct FlutterSerializedModel: Model, JSONValueHolder {
                     result[key] = modifiedArray
                 }
             } else if case .dateTime = field?.type,
-                    case .some(.string(let deserializedValue)) = values[key] {
-
+                      case .some(.string(let deserializedValue)) = values[key] {
                 result[key] = deserializedValue
             } else if case .date = field?.type,
-                    case .some(.string(let deserializedValue)) = values[key] {
-
+                      case .some(.string(let deserializedValue)) = values[key] {
                 result[key] = deserializedValue
             } else if case .time = field?.type,
-                    case .some(.string(let deserializedValue)) = values[key] {
-
+                      case .some(.string(let deserializedValue)) = values[key] {
                 result[key] = deserializedValue
             } else if case .timestamp = field?.type,
-                    case .some(.number(let deserializedValue)) = values[key] {
-                
+                      case .some(.number(let deserializedValue)) = values[key] {
                 result[key] = NSNumber(value: Int(deserializedValue) )
```

Comment:
These 4 branches seem to overlap with the implementation of 

```
private func extractJsonValue(key: String, value: JSONValue?, modelSchema: ModelSchema) -> Any??
```
Should we consolidate this logic as well?

Reply:
Great point I think we can just delete those cases as we call extractJsonValue at the end anyway.